expose-metrics/handler.go

```diff
@@ -0,0 +1,111 @@
+package function
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"regexp"
+	"strconv"
+
+	"github.com/openfaas/faas/gateway/metrics"
+)
+
+type Metrics struct {
+	Success int `json:"success"`
+	Failure int `json:"failure"`
+}
+
+// Handle a serverless request
+func Handle(req []byte) string {
+	fnName, err := parseFunctionName(string(req))
+	if err != nil {
+		log.Fatalf("Couldn't parse query %b, %t", req, err)
+	}
+
+	host := os.Getenv("prometheus_host")
+	envPort := os.Getenv("prometheus_port")
+	port, err := strconv.Atoi(envPort)
+	if err != nil {
+		log.Fatalf("Could not convert env-var prometheus_port to int. Env-var value: %s. Error: %t", envPort, err)
+	}
+
+	metricsQuery := metrics.NewPrometheusQuery(host, port, &http.Client{})
+
+	fnMetrics, err := getMetrics(fnName, metricsQuery)
+	if err != nil {
+		log.Fatalf("Couldn't get metrics from Prometheus for function %s, %t", fnName, err)
+	}
+
+	res, err := json.Marshal(fnMetrics)
+	if err != nil {
+		log.Fatalf("Couldn't marshal json %t", err)
+	}
+	return string(res)
+}
+
+func parseFunctionName(query string) (string, error) {
+	var fnName string
+
+	matcher := regexp.MustCompile("^/?(?:/system/)??(?:async-)?function/([^/?]+)([^?]*)")
```

Comment:
What’s this for?

Reply:
Currently I can see that function is taking body as an url and parsing the function name from it. Can we assume that the body of a request will be just a function name instead of url?

Reply:
If so I think we could just get rid of that function and its tests

Reply:
If we'll assume that body of request == functionName, should I do some checking/sanitising of the body before proceeding with request?

Reply:
You should not accept a post body at all, this should be passed via HTTP path or query string.

Reply:
Also the body of the request would/should never contain /system /async or the word function? Where did we get that from?